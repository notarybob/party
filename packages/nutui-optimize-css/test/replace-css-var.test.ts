import postcss from 'postcss'
import path from 'path'
import { describe, expect, it } from 'vitest'
import optimizeCss from '../dist/index.cjs'

const css = `
.nut-address-footer-btn {
  background: linear-gradient(135deg, var(--nutui-color-primary-stop-1, #f53d6d) 0%, var(--nutui-color-primary-stop-2, #fa2c19) 100%);
  color: var(--nutui-color-primary-text)
}
`
describe('@nutui/optimize-css', () => {
  it('optimize css', async () => {
    const a = await postcss([
      optimizeCss({
        cssVariables: {
          include: [path.join(__dirname, 'variables.scss')],
          exclude: ['--nutui-color-primary-text'],
          type: 'replace',
        },
      }),
    ]).process(css, { from: undefined })
    const optimizedCsss = a.css.toString()
    console.log(optimizedCsss)
    // @ts-ignore
    expect(optimizedCsss).toMatchSnapshot()
  })
})
