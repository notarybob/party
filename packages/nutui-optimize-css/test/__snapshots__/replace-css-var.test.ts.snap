// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`@nutui/optimize-css > optimize css 1`] = `
":root {
  --nutui-color-primary-text: blue;
}


.nut-address-footer-btn {
  background: linear-gradient(135deg, yellow 0%, #fa2c19 100%);
  color: blue;
  color: var(--nutui-color-primary-text)
}
"
`;
